api/src/services/roles.ts

```diff
@@ -350,12 +238,114 @@ export class RolesService extends ItemsService {
 		this.assertValidIpAccess(data);
 
 		try {
+			const increasedCounts: AccessTypeCount = {
+				admin: 0,
+				app: 0,
+				api: 0,
+			};
+
+			let increasedUsers = 0;
+
+			const existingRole:
+				| {
+						count: number | string;
+						admin_access: number | boolean | null;
+						app_access: number | boolean | null;
+				  }
+				| undefined = await this.knex
+				.count('directus_users.id', { as: 'count' })
+				.select('directus_roles.admin_access', 'directus_roles.app_access')
+				.from('directus_users')
+				.where('directus_roles.id', '=', key)
+				.andWhere('directus_users.status', '=', 'active')
+				.leftJoin('directus_roles', 'directus_users.role', '=', 'directus_roles.id')
+				.groupBy('directus_roles.admin_access', 'directus_roles.app_access')
+				.first();
```

Comment:
The query does not work for roles that previously had no users assigned, it returns `undefined`.
Effectively blocking the update of such a role at the moment when trying to add new users.

Reply:
To be addressed in https://github.com/directus/directus/pull/22642